
Curl's tests run from perl which invokes curl commands via system().
Perls's system() uses /bin/sh which on OmniOS is a link to the 32-bit
version of ksh93
ksh ends up removing the empty parameter to --[no]proxy without this patch.

Only in curl-7.57.0.patch/tests/data: Makefile
diff -wpruN '--exclude=*.orig' a~/tests/data/test1004 a/tests/data/test1004
--- a~/tests/data/test1004	1970-01-01 00:00:00
+++ a/tests/data/test1004	1970-01-01 00:00:00
@@ -36,7 +36,7 @@ http
 HTTP GET with empty proxy
  </name>
  <command>
-http://%HOSTIP:%HTTPPORT/1004 --proxy ""
+http://%HOSTIP:%HTTPPORT/1004 --proxy "''"
 </command>
 </client>
 
diff -wpruN '--exclude=*.orig' a~/tests/data/test1254 a/tests/data/test1254
--- a~/tests/data/test1254	1970-01-01 00:00:00
+++ a/tests/data/test1254	1970-01-01 00:00:00
@@ -33,7 +33,7 @@ Under condition using --proxy, override
 NO_PROXY=example.com
 </setenv>
 <command>
-http://somewhere.example.com/1254 --proxy http://%HOSTIP:%HTTPPORT --noproxy ""
+http://somewhere.example.com/1254 --proxy http://%HOSTIP:%HTTPPORT --noproxy "''"
 </command>
 <features>
 proxy
diff -wpruN '--exclude=*.orig' a~/tests/data/test1257 a/tests/data/test1257
--- a~/tests/data/test1257	1970-01-01 00:00:00
+++ a/tests/data/test1257	1970-01-01 00:00:00
@@ -34,7 +34,7 @@ http_proxy=http://%HOSTIP:%HTTPPORT
 NO_PROXY=example.com
 </setenv>
 <command>
-http://somewhere.example.com/1257 --noproxy ""
+http://somewhere.example.com/1257 --noproxy "''"
 </command>
 <features>
 proxy
